pkg/cloud/services/network/subnets.go

```diff
@@ -52,6 +52,8 @@ func (s *Service) reconcileSubnets() error {
 	subnets := s.scope.Subnets()
 	defer func() {
 		s.scope.SetSubnets(subnets)
+		s.scope.Debug("Reconciled subnets", "subnets", subnets)
+		conditions.MarkTrue(s.scope.InfraCluster(), infrav1.SubnetsReadyCondition)
```

Comment:
The deferred function also runs after errors and thus we don't know whether subnets are ready. The condition was previously set at the end of `reconcileSubnets` – why did this code get moved?

Reply:
A bit confused, I didn't add these lines